schemas/iso19139/src/main/plugin/iso19139/index-fields/index.xsl

```diff
@@ -471,10 +462,17 @@
             <!-- Index keyword characterString including multilingual ones
              and element like gmx:Anchor including the href attribute
              which may contains keyword identifier. -->
-            <xsl:for-each select="*[normalize-space() != '']|
+            <xsl:variable name="thesaurusField"
+                          select="concat('thesaurus_', replace($key, '[^a-zA-Z0-9]', ''))"/>
+
+            <xsl:element name="{$thesaurusField}Number">
```

Comment:
This field key is not very clear IMO. You end up having things like:
```json
"thesaurus_GEMETINSPIREthemesversion10" : "Zones de gestion, de restriction ou de réglementation et unités de déclaration",
"thesaurus_GEMETINSPIREthemesversion10Number" : "1",
```

The suffix looks to be part of the thesaurus identifier. Also the prefix `thesaurus_` does not clearly indicates that what follows are actual keywords. How about using `keywordsByThesaurus_...` and `keywordCountByThesaurus_...`? Just a suggestion.

Reply:
Maybe a more global review of fields should be done in https://github.com/geonetwork/core-geonetwork/blob/es/web/src/main/webapp/WEB-INF/data/config/index/records.json#L942 

We should maybe plan for this on ES branch.

This PR mainly address the move to ES6.4, new needs for analysis in Kibana dashboards and add authentication.

Reply:
I added a step in ES task list "Define index fields structures, naming and conventions"